### DISCUSSION TITLE
Symbols: Use of multiple sprites in mbs-files

### DISCUSSION THREAD

**Discussion by Sverre-Iversen** [ldproxy/ldproxy#858] [Configuration] [answered]
Because we use vector tiles from different providers, e.g. topographical maps from the National Mapping Authority and others, it should be possible to use multiple sprites (or use of "sprite atlas"?). If the OGC API Symbols draft does not yet comply with this it should be extended.

Today we have to combine all the sprites we need into one. This is not trivial, because we cannot generate the combined sprite automatically from the individual symbols from all providers. This is partly done manually by combining the individual sprites (png & json) into one. We also need to recalculate the offsets for the symbols based on the new "sub"-sprite offsets.

**Comment by cportele** [comment, marked as answer]
It is a current constraint of styles that there can only be one sprite per style (https://maplibre.org/maplibre-gl-js-docs/style-spec/sprite/).

However, this is changing. In the future multiple sprites per style will be supported (https://github.com/maplibre/maplibre-gl-js/pull/1805) by MapLibre.
